src/libraries/System.Text.Json/tests/Serialization/NumberHandlingTests.cs

```diff
@@ -0,0 +1,1414 @@
+
+            // Serialize
+            string serialized = JsonSerializer.Serialize(obj, s_optionReadAndWriteFromStr);
+
+            // Deserialize
+            obj = JsonSerializer.Deserialize<KeyValuePair<ulong?, List<float>>>(serialized, s_optionReadAndWriteFromStr);
+
+            // Test roundtrip
+            JsonTestHelper.AssertJsonEqual(serialized, JsonSerializer.Serialize(obj, s_optionReadAndWriteFromStr));
+        }
+
+        [Fact]
+        public static void Number_AsObjectWithParameterizedCtor_RoundTrip()
+        {
+            var obj = new MyClassWithNumbers(JsonNumberTestData.NullableULongs.LastOrDefault(), JsonNumberTestData.Floats);
+
+            // Serialize
+            string serialized = JsonSerializer.Serialize(obj, s_optionReadAndWriteFromStr);
+
+            // Deserialize
+            obj = JsonSerializer.Deserialize<MyClassWithNumbers>(serialized, s_optionReadAndWriteFromStr);
+
+            // Test roundtrip
+            JsonTestHelper.AssertJsonEqual(serialized, JsonSerializer.Serialize(obj, s_optionReadAndWriteFromStr));
+        }
+
+        private class MyClassWithNumbers
+        {
+            public ulong? Ulong { get; }
+            public List<float> ListOfFloats { get; }
+
+            public MyClassWithNumbers(ulong? @ulong, List<float> listOfFloats)
+            {
+                Ulong = @ulong;
+                ListOfFloats = listOfFloats;
+            }
+        }
+
+        [Fact]
+        public static void Number_AsObjectWithParameterizedCtor_PropHasAttribute()
+        {
+            string json = @"{""ListOfFloats"":[""1""]}";
+            // Strict handling on property overrides loose global policy.
+            Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<MyClassWithNumbers_PropsHasAttribute>(json, s_optionReadFromStr));
+
+            // Serialize
+            json = @"{""ListOfFloats"":[1]}";
+            MyClassWithNumbers_PropsHasAttribute obj = JsonSerializer.Deserialize<MyClassWithNumbers_PropsHasAttribute>(json);
+
+            // Number serialized as JSON number due to strict handling on property which overrides loose global policy.
+            Assert.Equal(json, JsonSerializer.Serialize(obj, s_optionReadAndWriteFromStr));
+        }
+
+        private class MyClassWithNumbers_PropsHasAttribute
+        {
+            [JsonNumberHandling(JsonNumberHandling.Strict)]
+            public List<float> ListOfFloats { get; }
+
+            public MyClassWithNumbers_PropsHasAttribute(List<float> listOfFloats)
+            {
+                ListOfFloats = listOfFloats;
+            }
+        }
+
+        [Fact]
+        public static void FloatingPointConstants_Pass()
+        {
+            // Valid values
+            PerformFloatingPointSerialization("NaN");
+            PerformFloatingPointSerialization("Infinity");
+            PerformFloatingPointSerialization("-Infinity");
```

Comment:
How about when parts of these strings are escaped?

Reply:
It would be good to add remarks to the API docs about reading escaped strings and the behavior of this setting.

Reply:
That is supported. Tests have been added and the docs will be updated.